### ISSUE TITLE
Skipping messages in outbox

### ISSUE THREAD

**Issue by dgtlmoon** [thundernest/k-9#4594]
```
03-09 09:36:20.967  3936  3956 I MessagingController: Running command 'sendPendingMessages', seq = 75 (background priority)
03-09 09:36:20.992  3936  3956 I MessagingController: Scanning folder 'K9MAIL_INTERNAL_OUTBOX' (7) for messages to send
03-09 09:36:20.994  3936  3956 V MessagingController: Skipping sending message K9LOCAL:1f994ee8-53e6-4521-a28d-812f43b71046
03-09 09:36:21.005  3936  3956 V MessagingController: Skipping sending message K9LOCAL:af201cb8-0110-41b1-acb0-d2ca49bbb948
03-09 09:36:21.011  3936  3956 I MessagingController:  Command 'sendPendingMessages' completed
03-09 09:36:21.297  3936  3985 D OpenGLRenderer: endAllActiveAnimators on 0x7a84a38300 (MenuPopupWindow$MenuDropDownListView) with handle 0x7a88deb500
```
### Expected behavior
2 messages in the Outbox, they both contain attachments, they should send

### Actual behavior
They do not send

### Steps to reproduce
Add a new email by sharing an image from the gallery as a new email, wait for send

### Environment
K-9 Mail version: 5.705

Android version:

Account type (IMAP, POP3, WebDAV/Exchange):
IMAPS+SMTPS

` Skipping sending message ` is a horrible message without any further details WHY the message was skipped... :(

```
03-09 09:36:20.967  3936  3956 I MessagingController: Running command 'sendPendingMessages', seq = 75 (background priority)
03-09 09:36:20.992  3936  3956 I MessagingController: Scanning folder 'K9MAIL_INTERNAL_OUTBOX' (7) for messages to send
03-09 09:36:20.994  3936  3956 V MessagingController: Skipping sending message K9LOCAL:1f994ee8-53e6-4521-a28d-812f43b71046
03-09 09:36:21.005  3936  3956 V MessagingController: Skipping sending message K9LOCAL:af201cb8-0110-41b1-acb0-d2ca49bbb948
03-09 09:36:21.011  3936  3956 I MessagingController:  Command 'sendPendingMessages' completed
03-09 09:36:21.297  3936  3985 D OpenGLRenderer: endAllActiveAnimators on 0x7a84a38300 (MenuPopupWindow$MenuDropDownListView) with handle 0x7a88deb500
```

**Comment by dgtlmoon** [comment]
I see...

```                    if (outboxState.getSendState() != SendState.READY) {
                        Timber.v("Skipping sending message " + message.getUid());
                        notificationController.showSendFailedNotification(account,
                                new MessagingException(message.getSubject()));
                        continue;
                    }
```

**Comment by dgtlmoon** [comment]
Sorry, wrong button hit

**Comment by cketti** [comment]
It could either be the number of allowed retries being exceeded or the server reported a permanent error. 

I created issue #4602 to improve error reporting on send failures and allow resolving send errors.

For now you'll have to create a new message to send. You can do so by going to the Outbox, long-pressing the message, then select "send again". Afterwards you can delete the message that failed to send.

**Comment by dgtlmoon** [comment]
@cketti OK Super, can you clarify "Server reported a permanent error", does this mean if it happens just one time, then that email wont send in the future? If it helps - I have full control of the mail server so I can test this situation in #4602 for you

I can confirm the `send again` trick worked, so "Send Again" actually means "Send a copy"

**Comment by cketti** [comment]
> can you clarify "Server reported a permanent error", 

Possible reasons:
- Message is larger than the server accepts (SMTP SIZE extension)
- SMTP response code 5xx on any request 

> does this mean if it happens just one time, then that email wont send in the future?

Correct. There should be an error notification letting you know that sending the message has failed.

**Comment by stuart12** [comment]
I have the same problem but as I was able to send the email (after using the badly named "Send Again" button and adding some text) I don't think that it was my email server that gave an error.  I think that at the moment I tried to send this message my tablet was not connected to the Internet.  It would be great if the "Send messages" button cleared errors, permanent or not, and tried to send the message again. My tablet only has WiFi so trying to send an email is free.

Could an extra option be added to each message to display why the most recent send failed?

**Comment by dgtlmoon** [comment]
the outbox handling is still horrible... :(

my postconf is huge

```
root@mailer:/var/log# postconf | grep message_size_limit
message_size_limit = 152428800
```

but it's still reporting "Message too large for server", the only way I can resend is to forward the email (no retry button, does not retry on a new sync)

even WORSE is that the folder counter for OUTBOX is 0... so you don't even know that you were not able to send an email == data loss = wrong expectation (maybe you assume that the other guy got the email, but he did not, so you look like an idiot)

**Comment by dgtlmoon** [comment]
Please I'm begging for .. [#1758]

**Comment by cketti** [comment]
K-9 Mail 6.303 will show an error notification each time the app tries to send messages and there's a message with an error left in the Outbox.

**Comment by kevgrig** [comment]
> For now you'll have to create a new message to send. You can do so by going to the Outbox, long-pressing the message, then select "send again". Afterwards you can delete the message that failed to send.

There was no such "send again" option for me (Android) but I was able to edit the message and change it to draft, then go to drafts, edit it, and then re-send.
